### ISSUE TITLE
Suggestion: make the cabal completion more useful when `--` is present

### ISSUE THREAD

**Issue by Ptival** [fish-shell/fish-shell#10589]
Currently, the cabal completion list cabal options from `cabal ... --list-options` no matter what.

However, when trying to complete:
```
cabal run tool -- /path/to/<TAB>
```

This is quite annoying as it takes a while to list cabal options that are completely irrelevant in this prompt context: we're no longer building cabal options, these are arguments to the underlying tool.

Would it be possible to get back to `ls` completion when a `--` is present in the argument list?
I had a quick look at the repo, and saw code like `if contains -- $cmd`, is this the intended way of dealing with such cases? Is this documented somewhere?

I could try and create this patch if this seems acceptable.

**Comment by mqudsi** [comment]
No, `contains -- $cmd` is to tell `contains` not to parse any `$cmd` contents that begin with a `-` or `--` as a flag (same as in your `cabal` example above!)

To use `contains` to test for `--` in `$cmd`, you would need to use it twice, once to stop flag processing and again as the payload you are actually seeking, so `contains -- -- $cmd`.

I think simply adding `-n 'not contains -- -- (commandline -xpc)'` to each of the existing completions would accomplish what you need, but I am now thinking in terms of what can we do to make this easier out-of-the-box.
